corehq/util/metrics/__init__.py

```diff
@@ -267,6 +269,8 @@ def stop(self, name=None):
             self._metric, self.duration, bucket_tag='duration',
             buckets=self._timing_buckets, bucket_unit='s', tags=self._tags
         )
+        if self._errored:
```

Comment:
Note that an error will be recorded twice - we'll still record the duration, but add a second count for the failure.  At present, I'm doing this for all `metrics_histogram_timer` usages, but I could be persuaded to make it an optional argument.